pkg/build.go

```diff
@@ -51,11 +53,69 @@ type Target interface {
 	Export() []EnvVar
 }
 
+// ImageReference defines the reference to a docker image
+type ImageReference struct {
+	Registry   string `json:"registry"`
+	Repository string `json:"repository"`
+	Tag        string `json:"tag,omitempty"`
+	Digest     string `json:"digest"`
+	reference  reference.Reference
+}
+
+// NewImageReference parse a path of a image and create a ImageReference object
```

Comment:
NIT: some wording throughout docs. (parse**s**, create**s**). Same comment applies for `String()`